app/models/employee.rb

```diff
@@ -0,0 +1,26 @@
+class Employee < ApplicationRecord
+  has_many :addresses, dependent: :destroy
+
+  accepts_nested_attributes_for :addresses
+
+  enum :gender, %i[male female]
+  HOBBY = %i[Dancing Singing Travelling].freeze
+
+  validates :employee_name, :email, :password, :gender, :hobbies, :address, :mobile_number, :birth_date, :document, presence: true
+  validates :mobile_number, length: { is: 10 }
+  validates :employee_name, :email, uniqueness: true
+  validates :password, length: { minimum: 6 }
+
+  before_create :create_user
+  before_destroy :destroy_user
+
+  private
+
+  def create_user
+    User.create(username: employee_name, email: email, password: password, password_confirmation: password, role: "Employee") unless User.exists?(username: employee_name)
```

Comment:
As you have made changes in the enum role:`app/models/user.rb` in this file. You need to make the relevant changes here.